internal/provider/user_data_source.go

```diff
@@ -0,0 +1,163 @@
+		Description: "A user belonging to a Segment Workspace.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Required:    true,
+				Description: "The unique identifier for this user.",
+			},
+			"name": schema.StringAttribute{
+				Description: "The human-readable name of this user.",
+				Computed:    true,
+			},
+			"email": schema.StringAttribute{
+				Description: "The email address associated with this user.",
+				Computed:    true,
+			},
+			"permissions": schema.SetNestedAttribute{
+				Description: "The permissions associated with this user.",
+				Computed:    true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"role_id": schema.StringAttribute{
+							Description: "The id of the role associated with this permission.",
+							Computed:    true,
+						},
+						"resources": schema.SetNestedAttribute{
+							Description: "The resources associated with this permission.",
+							Computed:    true,
+							NestedObject: schema.NestedAttributeObject{
+								Attributes: map[string]schema.Attribute{
+									"id": schema.StringAttribute{
+										Description: "The id of this resource.",
+										Computed:    true,
+									},
+									"type": schema.StringAttribute{
+										Description: "The type for this resource.",
+										Computed:    true,
+									},
+									"labels": schema.SetNestedAttribute{
+										Description: "The labels that further refine access to this resource. Labels are exclusive to Workspace-level permissions.",
+										Computed:    true,
+										NestedObject: schema.NestedAttributeObject{
+											Attributes: map[string]schema.Attribute{
+												"key": schema.StringAttribute{
+													Description: "The key that represents the name of this label.",
+													Computed:    true,
+												},
+												"value": schema.StringAttribute{
+													Description: "The value associated with the key of this label.",
+													Computed:    true,
+												},
+												"description": schema.StringAttribute{
+													Description: "An optional description of the purpose of this label.",
+													Computed:    true,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (d *userDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var config models.UserDataSourceState
+	diags := req.Config.Get(ctx, &config)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	id := config.ID.ValueString()
+	if id == "" {
+		resp.Diagnostics.AddError("Unable to read user", "ID is empty")
+
+		return
+	}
+
+	out, body, err := d.client.IAMUsersApi.GetUser(d.authContext, id).Execute()
+	if body != nil {
+		defer body.Body.Close()
+	}
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to read User",
```

Comment:
Just for consistency, the other error messages in this file used a lower case.
```suggestion
			"Unable to read user",
```